src/wallet/wallet.cpp

```diff
@@ -2726,7 +2741,8 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
     // enough, that fee sniping isn't a problem yet, but by implementing a fix
     // now we ensure code won't be written that makes assumptions about
     // nLockTime that preclude a fix later.
-    txNew.nLockTime = chainActive.Height();
+    const Optional<int> tip_height = locked_chain.getHeight();
+    txNew.nLockTime = tip_height ? *tip_height : std::numeric_limits<uint32_t>::max();
```

Comment:
Probably out-of-scope nit: I wish the uint32_t max was instead a named constant that had meaning relative to use with nLockTime (e.g. `IGNORE_NLOCKTIME`).